src/microbe_stage/editor/CellEditorComponent.cs

```diff
@@ -1930,6 +1944,42 @@ private void StartAutoEvoPrediction()
         return ProcessSystem.ComputeCompoundFillTimes(compoundBalanceData, nominalStorage, specificStorages);
     }
 
+    private void HandleProcessList(EnergyBalanceInfo energyBalance, BiomeConditions biome)
+    {
+        var processes = new List<TweakedProcess>();
+
+        // Empty list to later fill
+        var processStatistics = new List<ProcessSpeedInformation>();
+
+        ProcessSystem.ComputeActiveProcessList(editedMicrobeOrganelles, ref processes);
+
+        float consumptionProductionRatio = energyBalance.TotalConsumption / energyBalance.TotalProduction;
+
+        foreach (var process in processes!)
+        {
+            var singleProcess = ProcessSystem.CalculateProcessMaximumSpeed(process, biome, CompoundAmountType.Current);
+
+            // If produces more ATP than consumes, lower down production for inputs and for outputs,
+            // otherwise use maximum production values
+            if (consumptionProductionRatio < 1.0f)
+            {
+                foreach (var input in singleProcess.Inputs)
```

Comment:
If I remember right the Inputs property filters out environmental inputs. Does this not adjusting those match the behaviour that is in the during gameplay process panel?

Reply:
Gameplay process panel shows environment inputs at most as @ multiplier at the end of equation.

Reply:
Well yes, but are those affected by the actual process speed (I cannot remember off the top of my head)? If not then this code is correct.

Reply:
At least not by Rate

Reply:
Looking at the stuff in the process system it needs the environmental compound usage that is reported is just the raw values and not adjusted.

Reply:
What do you mean by that?

Reply:
If you modify a dictionary that you are iterating with foreach and exit the loop, it won't cause an exception.

Collection modified exception is only triggered on the next loop iteration. The same applies here, as there's always just a single item to iterate: the code exits the loop after modifying the dictionary so there is no error. But if anyone ever adds a bio process to the game that takes in 2 non-environmental inputs, then they are going to have a really bad time trying to debug this.